src/service/ratelimit.go

```diff
@@ -126,6 +141,37 @@ func (this *service) shouldRateLimitWorker(
 			finalCode = descriptorStatus.Code
 		}
 	}
+	if this.responseHeadersEnabled {
+		now := this.clock.Now().Unix()
+		var limitingDescriptor *pb.RateLimitResponse_DescriptorStatus
+		limitCount := 0
+		for _, descriptor := range responseDescriptorStatuses {
+			if descriptor.CurrentLimit == nil {
+				continue
+			}
+			limitCount++
+			if limitingDescriptor == nil ||
+				descriptor.LimitRemaining < limitingDescriptor.LimitRemaining ||
+				descriptor.LimitRemaining == limitingDescriptor.LimitRemaining &&
+					calculateReset(descriptor, now) > calculateReset(limitingDescriptor, now) {
+				limitingDescriptor = descriptor
+			}
+		}
+		if limitCount == 1 {
+			response.Headers = []*core.HeaderValue{
+				limitHeader(limitingDescriptor),
+				remainingHeader(limitingDescriptor),
+				resetHeader(limitingDescriptor, now),
+			}
+		} else if limitCount > 1 {
+			// If there is more than one limit, then picking one of them for the "X-RateLimit-Limit"
```

Comment:
If we pick min for remaining, shouldn't we pick the accompanying Limit to fill the limitHeader?